.travis.yml

```diff
@@ -49,60 +19,161 @@ addons:
+          fi
+        - if [[ $NO_DEBUGGER == 1 ]]; then xdebug-disable; fi
+        - |
+          if [[ $PHPDBG != 1 ]]; then
+            bin/infection $INFECTION_FLAGS;
+          else
+            phpdbg -qrr bin/infection $INFECTION_FLAGS;
+          fi
+
+      after_success:
+        - bash <(curl -s https://codecov.io/bash)
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.0
+      env: NO_DEBUGGER=1
+
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.0
+      env: PHPDBG=1
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.1
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.1
+      env: NO_DEBUGGER=1
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.1
+      env: PHPDBG=1
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.1
+      env: SYMFONY_VERSION="^4.0"
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.1
+      env: SYMFONY_VERSION="^4.0" PHPDBG=1
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.2
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.2
+      env: NO_DEBUGGER=1
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.2
+      env: PHPDBG=1
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.1
+      env: SYMFONY_VERSION="^4.0"
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: 7.2
+      env: SYMFONY_VERSION="^4.0" PHPDBG=1
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
+      php: nightly
+
+      script:
+        # We don't have xdebug on the nightly build, so we always run with phpdbg.
+        - make build/bin/infection.phar;
+        - ./tests/e2e_tests build/bin/infection.phar;
+        - phpdbg -qrr $PHPUNIT_BIN
+        - phpdbg -qrr bin/infection $INFECTION_FLAGS
+
+    -
+      stage: Deploy
+      php: 7.2
+      install:
+        - composer install
+      deploy:
+        provider: releases
+        api_key:
+          secure: JtdvJBE2ARf6VWN5KlXAburz3FZMriiXBHEdeIocHePBAzgYW7tVMksJR5N3azrE3KjhftMEukXtA1k5VfAGdLCSm6nKGsZwolkd24ryG7PYNuAdFWfbau8QQXn6aLFesF2lkTq89+PqxAdVjwXNgFA7ble3vL66Ogabff6gov6QwT9vDrTqwLmWNGwbt9QB32IE9yVPjW/3lk4MAkMHtKDr2DElyeT2CRRuPuUCA+zNwoGG95wqLAD/RwjsqR3GjotzDKrldqog9jU2OQptUc3BgdKlYR1AzdknxJYnT7a5CRVdmSOEfD9IrDqlYEzn9pHav+Sk4KD2KNKsALQGk/w16BqncN2uyOexUTzXvnDd27G4ZYy+1OsXMdYMhF5ffwj7EqNLm8XfLPtHHV0/FUJY9kkrbf9+wQVjDQmMJLLIiBSt3vuY0ya5SVW8thtjlF2j67smamNe4hDhditBipto+9Y5vI4AYhcm4CQlqP8BVbw3/LgGAl/dJR5ohM7PFZH3EUE1AojBaEXfUymlICdFVZFJbB0D/AM8jSqNmWBQu9QX3MURhT5xLSwc+kir2izQaEo/GQz79AifqkUJgSCzpRUJntCN8A7yk1uxDZDONbINF3ReuF+nqu7GMksgKZLCGbOIaBl9of24K5BeTqxgcEykiHFFY7X9QdI3Dyo=
+        file:
+          - build/bin/infection.phar
+          - build/bin/infection.phar.pubkey
+        skip_cleanup: true
+        on:
+          tags: true
+          repo: infection/infection
+
+
+allow_failures:
```

Comment:
As @borNfreee pointed out, this should be a under `jobs`.

Also see https://github.com/travis-ci/travis-ci/issues/7789.